test/jdk/java/net/ipv6tests/UdpTest.java

```diff
@@ -138,26 +139,27 @@ static void test2 () throws Exception {
         s1 = new DatagramSocket ();
         s2 = new DatagramSocket ();
         s1.setSoTimeout (4000);
-        long t1 = System.currentTimeMillis();
+        long t1 = System.nanoTime();
         try {
             s1.receive (new DatagramPacket (new byte [128], 128));
             throw new Exception ("expected receive timeout ");
         } catch (SocketTimeoutException e) {
         }
-        checkTime (System.currentTimeMillis() - t1, 4000);
+        final long expectedTimeInNanos = TimeUnit.SECONDS.toNanos(4);
+        checkTime (System.nanoTime() - t1, expectedTimeInNanos);
```

Comment:
Actually, maybe it would be better to keep the expectedTime in milliseconds here, and convert `System.nanoTime() - t1` to milliseconds before calling checkTime. We obviously don't need the nano second precision, and milliseconds are more human readable than nanoseconds when looking at the log.